compiler/plugins/input/Torch/InputConversion/ConvertTorchUnstructuredToLinalgExt.cpp

```diff
@@ -163,6 +171,349 @@ struct FftRfftOpConversion
+    auto toBuiltinTensor = [&](Value torchTensor) -> Value {
+      auto torchType =
+          cast<torch::Torch::ValueTensorType>(torchTensor.getType());
+      return torch::TorchConversion::ToBuiltinTensorOp::create(
+          rewriter, loc, torchType.toBuiltinTensor(), torchTensor);
+    };
+
+    Value builtinQuery = toBuiltinTensor(query);
+    Value builtinKey = toBuiltinTensor(key);
+    Value builtinValue = toBuiltinTensor(value);
+
+    // Declare common types for mask and score modification regions
+    Type i32Type = rewriter.getI32Type();
+    Type si32Type =
+        IntegerType::get(rewriter.getContext(), 32, IntegerType::Signed);
+    auto scalarTensorType = RankedTensorType::get({}, floatType);
+    auto i32ScalarTensorType = RankedTensorType::get({}, i32Type);
+    auto boolScalarTensorType = RankedTensorType::get({}, rewriter.getI1Type());
+    auto torchScalarType = rewriter.getType<torch::Torch::ValueTensorType>(
+        ArrayRef<int64_t>{}, floatType);
+    auto torchI32ScalarType = rewriter.getType<torch::Torch::ValueTensorType>(
+        ArrayRef<int64_t>{}, si32Type);
+    auto torchBoolScalarType = rewriter.getType<torch::Torch::ValueTensorType>(
+        ArrayRef<int64_t>{}, rewriter.getI1Type());
+
+    // Create mask if mask_mod_fn is provided
+    Value mask;
+    if (maskModSymbol) {
+      // Create mask tensor [B, H, M, N] with values 0.0 (attend) or -inf (mask)
+      SmallVector<int64_t> maskShape = {batch, numHeads, seqLenQ, seqLenKV};
+      SmallVector<Value> maskDynSizes;
+
+      // Handling dynamic dimensions
+      for (int i = 0; i < 4; ++i) {
+        if (maskShape[i] == torch::Torch::kUnknownSize) {
+          Value idx = arith::ConstantIndexOp::create(rewriter, loc, i);
+          Value dim =
+              (i < 2) ? tensor::DimOp::create(rewriter, loc, builtinQuery, idx)
+              : (i == 2)
+                  ? tensor::DimOp::create(rewriter, loc, builtinQuery, idx)
+                  : tensor::DimOp::create(
+                        rewriter, loc, builtinKey,
+                        arith::ConstantIndexOp::create(rewriter, loc, 2));
+          maskDynSizes.push_back(dim);
+        }
+      }
+
+      Value maskTensor = tensor::EmptyOp::create(rewriter, loc, maskShape,
+                                                 floatType, maskDynSizes);
+      // Create linalg.generic to materialize mask
+      SmallVector<AffineMap> maskMaps;
+      maskMaps.push_back(
+          AffineMap::getMultiDimIdentityMap(4, ctx)); // output map
+
+      SmallVector<utils::IteratorType> iteratorTypes(
+          4, utils::IteratorType::parallel);
+
+      Value zero = arith::ConstantOp::create(
+          rewriter, loc, floatType, rewriter.getFloatAttr(floatType, 0.0));
+      Value negInf = arith::ConstantOp::create(
+          rewriter, loc, floatType,
+          rewriter.getFloatAttr(floatType,
+                                -std::numeric_limits<double>::infinity()));
```

Comment:
For zero, you can use `arith::ConstantFloatOp::create(..., floatType, 0)`.

For negInf, you should try this method: https://github.com/llvm/llvm-project/blob/3a2de951a40877e3d17aa87abb35565da1040612/llvm/include/llvm/ADT/APFloat.h#L1080-L1084

`double` != floatType. The precision can mismatch.